modules/topicsFpdModule.js

```diff
@@ -0,0 +1,80 @@
+import {logError, logWarn, mergeDeep} from '../src/utils.js';
+import {getRefererInfo} from '../src/refererDetection.js';
+import {submodule} from '../src/hook.js';
+import {GreedyPromise} from '../src/utils/promise.js';
+
+const TAXONOMIES = {
+  // map from topic taxonomyVersion to IAB segment taxonomy
+  '1': 600
+}
+
+function partitionBy(field, items) {
+  return items.reduce((partitions, item) => {
+    const key = item[field];
+    if (!partitions.hasOwnProperty(key)) partitions[key] = [];
+    partitions[key].push(item);
+    return partitions;
+  }, {});
+}
+
+export function getTopicsData(name, topics, taxonomies = TAXONOMIES) {
+  return Object.entries(partitionBy('taxonomyVersion', topics))
+    .filter(([taxonomyVersion]) => {
+      if (!taxonomies.hasOwnProperty(taxonomyVersion)) {
+        logWarn(`Unrecognized taxonomyVersion from Topics API: "${taxonomyVersion}"; topic will be ignored`);
+        return false;
+      }
+      return true;
+    }).flatMap(([taxonomyVersion, topics]) =>
+      Object.entries(partitionBy('modelVersion', topics))
+        .map(([modelVersion, topics]) => {
+          const datum = {
+            ext: {
+              segtax: taxonomies[taxonomyVersion],
+              segclass: modelVersion
+            },
+            segment: topics.map((topic) => ({id: topic.topic.toString()}))
+          };
+          if (name != null) {
+            datum.name = name;
+          }
+          return datum;
+        })
+    );
+}
+
+export function getTopics(doc = document) {
+  let topics = null;
+  try {
+    if ('browsingTopics' in doc && doc.featurePolicy.allowsFeature('browsing-topics')) {
+      topics = GreedyPromise.resolve(doc.browsingTopics());
+    }
+  } catch (e) {
+    logError('Could not call topics API', e);
+  }
+  if (topics == null) {
+    topics = GreedyPromise.resolve([]);
+  }
+  return topics;
+}
+
+const topicsData = getTopics().then((topics) => getTopicsData(getRefererInfo().domain, topics));
```

Comment:
It seems we should potentially use the domain of the url in https://developer.mozilla.org/en-US/docs/Web/API/Document/currentScript 

Pubishers may wish to indicate the breadth of the footprint here, eg hearst.com instead of somemagazine.com or cafemedia.com instead of cafedelites.com

Reply:
I'm not sure I understand - do you mean that topics will return different results if you use `<script src="//rubicon.com/prebid.js" />` compared `<script src="//appnexus.com/prebid.js" />`?  that seems insane.

Reply:
That's how it is described

Reply:
So it appears there's been some clarification:

"A design goal of the Topics API is to enable interest-based advertising without the sharing of information to more entities than is currently possible with third-party cookies. The Topics API proposes that topics can only be returned for API callers that have already observed them, within a limited timeframe.

Key Term
A Topics API caller is the entity that calls the document.browsingTopics() JavaScript method, and will use the topics returned by the method to help select relevant ads. Typically, a call to document.browsingTopics() would be from code included in a site from a third party such as an adtech platform. The browser determines the caller from the site of the current document. So, if you're a third party on a page, make sure you call the API from an iframe that your site owns."

This suggests only if you call the api from inside a frame and do the post-message will you get different behavior

Let's keep that for a follow up

Reply:
@dgirardi testing successful